crates/ethereum-forks/src/forkcondition.rs

```diff
@@ -108,3 +116,182 @@ impl ForkCondition {
+        let fork_condition = ForkCondition::Timestamp(12345);
+        assert!(
+            fork_condition.transitions_at_timestamp(12345, 12344),
+            "The condition should transition at timestamp 12345"
+        );
+
+        // Test if the condition does not transition if the parent timestamp is already the same
+        assert!(
+            !fork_condition.transitions_at_timestamp(12345, 12345),
+            "The condition should not transition if the parent timestamp is already 12345"
+        );
+        // Test with earlier timestamp
+        assert!(
+            !fork_condition.transitions_at_timestamp(123, 122),
+            "The condition should not transition if the parent timestamp is earlier"
+        );
+    }
+
+    #[test]
+    fn test_active_at_head() {
+        let head = Head {
+            hash: Default::default(),
+            number: 10,
+            timestamp: 12345,
+            total_difficulty: U256::from(1000),
+            difficulty: U256::from(100),
+        };
+
+        // Test if the condition activates based on block number
+        let fork_condition = ForkCondition::Block(10);
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head block number"
+        );
+        let fork_condition = ForkCondition::Block(11);
+        assert!(
+            !fork_condition.active_at_head(&head),
+            "The condition should not be active at the given head block number"
+        );
+
+        // Test if the condition activates based on timestamp
+        let fork_condition = ForkCondition::Timestamp(12345);
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head timestamp"
+        );
+        let fork_condition = ForkCondition::Timestamp(12346);
+        assert!(
+            !fork_condition.active_at_head(&head),
+            "The condition should not be active at the given head timestamp"
+        );
+
+        // Test if the condition activates based on total difficulty and block number
+        let fork_condition =
+            ForkCondition::TTD { fork_block: Some(9), total_difficulty: U256::from(900) };
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head total difficulty"
+        );
+        let fork_condition =
+            ForkCondition::TTD { fork_block: None, total_difficulty: U256::from(900) };
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head total difficulty as the block number is unknown"
+        );
+        let fork_condition =
+            ForkCondition::TTD { fork_block: Some(11), total_difficulty: U256::from(900) };
+        assert!(
+            !fork_condition.active_at_head(&head),
+            "The condition should not be active as the block is higher than head"
+        );
```

Comment:
This condition should be active because the ttd is higher

Reply:
Fixed